content/reference/sql/version-control/dolt-sql-functions.md

```diff
@@ -798,3 +799,170 @@ With result of single row:
+3. The second alter statement is the `ALTER TABLE...ADD` statement that adds the `color` column to the table.
+4. The `from_table_name` and `to_table_name` columns show the table name at the start and end of the diff. These columns are the same because the table was not renamed during the diff.
+5. Both the `from_create_statement` and `to_create_statement` columns show the full `CREATE TABLE` statement for the table at the start and end of the diff.
+
+We invoked `DOLT_SCHEMA_DIFF()` with branch names, but we could have used any revision specifier. For example, we could have used commit hashes or tag names, and would have gotten the same results.
+
+Using tags or commit hashes:
+
+```sql
+select * from dolt_schema_diff('v1', 'v1.1', 'inventory');
+select * from dolt_schema_diff('tjj1kp2mnoad8crv6b94mh4a4jiq7ab2', 'v391rm7r0t4989sgomv0rpn9ue4ugo6g', 'inventory');
+```
+
+So far, we have always supplied the last parameter, table name, but this is optional. If we omit the table name, we will get back a row for every table with a schema diff between the two revisions.
+
+In the current example, we have only one table with a schema diff, so the below queries return the same results as the previous query.
+
+```sql
+select * from dolt_schema_diff('main', 'feature_branch');
+select * from dolt_schema_diff('v1', 'v1.1');
+select * from dolt_schema_diff('tjj1kp2mnoad8crv6b94mh4a4jiq7ab2', 'v391rm7r0t4989sgomv0rpn9ue4ugo6g');
+```
+
+We can use the three dot diff to get the schema diff starting at the last common commit. In this example, the last common commit is the commit where the `feature_branch` was created from `main`.
+
+```sql
+select * from dolt_schema_diff('main...feature_branch');
+select * from dolt_schema_diff('v1...v1.1');
+select * from dolt_schema_diff('tjj1kp2mnoad8crv6b94mh4a4jiq7ab2...v391rm7r0t4989sgomv0rpn9ue4ugo6g');
+```
+
+Finally, we can flip the order of the revisions to get the schema diff in the opposite direction.
+
+```sql
+select * from dolt_schema_diff('feature_branch', 'main');
+```
+
+The above query will produce this output:
+
+```text
++-----------------+---------------+-------------------------------------------------------------------+-------------------------------------------------------------------+---------------------------------------------+
+| from_table_name | to_table_name | from_create_statement                                             | to_create_statement                                               | alter_statement                             |
++-----------------+---------------+-------------------------------------------------------------------+-------------------------------------------------------------------+---------------------------------------------+
+| inventory       | inventory     | CREATE TABLE `inventory` (                                        | CREATE TABLE `inventory` (                                        | ALTER TABLE `inventory` DROP `color`;       |
```

Comment:
Not docs related, but just looking at the function output... I'm wondering if we should be including the alter statements here. The output is just a little awkward with the same, long create table statements repeated on each line. Might be a cleaner interface to just have the table schemas, without repeating them, then do separate calls to `dolt_patch` to get the statements to run if needed. 

Would that have worked for the CLI migration you need this for? I know it would mean additional calls to the sql-server for the alter statements.

Reply:
Yeah, on reflection, `dolt_patch` could have given us the required information. We would have ended up with a few more calls, and a bit of a refactor, but it would have worked. 😐

Reply:
This function is new enough that we could still alter it without impacting customers (although we'd still want to call it out as a minor version just in case). It'll be **much** harder to change it in the future and we'll be stuck with this interface, so if we're not fully happy with it now, I think it's worth a couple hours to shift towards using `dolt_patch`. That's my vote, but if @macneale4 or @zachmu have a strong aversion to that, I could probably be convinced to just move on.

Reply:
PR opened to drop the `alter_statement` column: https://github.com/dolthub/dolt/pull/6270

Reply:
PR updated to match latest functionality, but the latest version of dolt is not deployed to DoltHub, so the embedded examples won't match the documentations.
PR converted to Draft and holding off on merging until the embedded examples match the new documentation.

Reply:
Latest version deployed to DoltHub and the embedded doc link is working as expected:
https://www.dolthub.com/repositories/dolthub/docs_examples/embed/main?active=Tables&q=SELECT+*%0AFROM+dolt_schema_diff%28%27schema_diff_v1%27%2C+%27schema_diff_v2%27%29%3B%0A